### ISSUE TITLE
Could not convert string to boolean

### ISSUE THREAD

**Issue by MarvinKlar** [wp-net/WordPressPCL#147]
When calling

`Dim wpcTest As New WordPressClient(....)
....
Dim lstTaxonomies As IEnumerable(Of Models.Taxonomy) = wpcTest.Taxonomies.Query(New TaxonomiesQueryBuilder With {
.Context = Models.Context.Edit
}, True).Result`

I get the following error:
`System.AggregateException: Mindestens ein Fehler ist aufgetreten. ---> Newtonsoft.Json.JsonReaderException: Could not convert string to boolean: manage_categories. Path 'category.capabilities.manage_terms', line 1, position 101.
   bei Newtonsoft.Json.JsonReader.ReadBooleanString(String s)
   bei Newtonsoft.Json.JsonTextReader.ReadAsBoolean()
   bei Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateDictionary(IDictionary dictionary, JsonReader reader, JsonDictionaryContract contract, JsonProperty containerProperty, String id)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateDictionary(IDictionary dictionary, JsonReader reader, JsonDictionaryContract contract, JsonProperty containerProperty, String id)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   bei Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   bei Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   bei Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
   bei WordPressPCL.Utility.HttpHelper.<GetRequest>d__19`1.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei WordPressPCL.Client.Taxonomies.<Query>d__7.MoveNext()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   bei System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   bei System.Threading.Tasks.Task`1.get_Result()
   bei Test.APPStart.VB$StateMachine_2_GO.MoveNext() in MyClass.vb:Zeile 496.
---> (Interne Ausnahme #0) Newtonsoft.Json.JsonReaderException: Could not convert string to boolean: manage_categories. Path 'category.capabilities.manage_terms', line 1, position 101.
   bei Newtonsoft.Json.JsonReader.ReadBooleanString(String s)
   bei Newtonsoft.Json.JsonTextReader.ReadAsBoolean()
   bei Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateDictionary(IDictionary dictionary, JsonReader reader, JsonDictionaryContract contract, JsonProperty containerProperty, String id)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateDictionary(IDictionary dictionary, JsonReader reader, JsonDictionaryContract contract, JsonProperty containerProperty, String id)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   bei Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   bei Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   bei Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   bei Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
   bei WordPressPCL.Utility.HttpHelper.<GetRequest>d__19`1.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei WordPressPCL.Client.Taxonomies.<Query>d__7.MoveNext()<---
`

Is this a bug?

**Comment by polushinmk** [comment]
Hi @MarvinKlar !
Could you provide a full JSON responce? or a part of it with 'category.capabilities.manage_terms' value

**Comment by MarvinKlar** [comment]
Hi @polushinmk,

where can I find this information? I can't see this information in Visual Studio.

**Comment by polushinmk** [comment]
You can intercept any requests with fiddler.

1. Start fiddler
2. Launch your code with problem
3. Find request to your site in fiddler, copy JSON response and paste it here :)

**Comment by polushinmk** [comment]
@ThomasPe seems that here is another strange thing in WP response.
**Capabilities** and **ExtraCapabilities** properties in User response is a dictionary where value can be bool (95%) or string (5%). 
I see 2 options to fix it:
1. change type from `Dictionary<string, bool>` to `Dictionary<string,object>`
2. write own custom JSON converter

**Comment by ThomasPe** [comment]
Do you have the JSON response somewhere? Is this selfhosted or WordPress.com?

**Comment by polushinmk** [comment]
selfhosted
```json
"capabilities": {
    "switch_themes": true,
    "edit_themes": true,
    "activate_plugins": true,
    "edit_plugins": true,
    "edit_users": true,
    "edit_files": true,
    "manage_options": true,
    "moderate_comments": true,
    "manage_categories": true,
    "manage_links": true,
    "upload_files": true,
    "import": true,
    "unfiltered_html": true,
    "edit_posts": true,
    "edit_others_posts": true,
    "edit_published_posts": true,
    "publish_posts": true,
    "edit_pages": true,
    "read": true,
    "level_10": true,
    "level_9": true,
    "level_8": true,
    "level_7": true,
    "level_6": true,
    "level_5": true,
    "level_4": true,
    "level_3": true,
    "level_2": true,
    "level_1": true,
    "level_0": true,
    "edit_others_pages": true,
    "edit_published_pages": true,
    "publish_pages": true,
    "delete_pages": true,
    "delete_others_pages": true,
    "delete_published_pages": true,
    "delete_posts": true,
    "delete_others_posts": true,
    "delete_published_posts": true,
    "delete_private_posts": true,
    "edit_private_posts": true,
    "read_private_posts": true,
    "delete_private_pages": true,
    "edit_private_pages": true,
    "read_private_pages": true,
    "delete_users": true,
    "create_users": true,
    "unfiltered_upload": true,
    "edit_dashboard": true,
    "update_plugins": true,
    "delete_plugins": true,
    "install_plugins": true,
    "update_themes": true,
    "install_themes": true,
    "update_core": true,
    "list_users": true,
    "remove_users": true,
    "promote_users": true,
    "edit_theme_options": true,
    "delete_themes": true,
    "export": true,
    "manage_polls": true,
    "NextGEN Manage tags": true,
    "NextGEN Manage others gallery": true,
    "MailPress_edit_dashboard": true,
    "MailPress_manage_options": true,
    "MailPress_edit_mails": true,
    "MailPress_edit_others_mails": true,
    "MailPress_send_mails": true,
    "MailPress_delete_mails": true,
    "MailPress_archive_mails": true,
    "MailPress_mail_custom_fields": true,
    "MailPress_switch_themes": true,
    "MailPress_edit_users": true,
    "MailPress_delete_users": true,
    "MailPress_user_custom_fields": true,
    "MailPress_manage_addons": true,
    "MailPress_manage_autoresponders": true,
    "MailPress_manage_newsletters": true,
    "MailPress_test_newsletters": true,
    "picasa_dialog": true,
    "wysija_newsletters": true,
    "wysija_subscribers": true,
    "wysija_config": true,
    "wysija_theme_tab": true,
    "wysija_style_tab": true,
    "wysija_stats_dashboard": true,
    "aiosp_manage_seo": true,
    "see_snap_box": true,
    "make_snap_posts": true,
    "wpseo_bulk_edit": true,
    "wpseo_manage_options": true,
    "haveown_snap_accss": true,
    "administrator": "1"
  },
  "extra_capabilities": {
    "administrator": "1"
  }

**Comment by ThomasPe** [comment]
thanks! I think a custom converter would be the better way here since the strings (which I guess are there due to bad plugin coding) still represent boolean values in the end.
Also, at some point it should be up to the Plugin devs to fix their inconsistencies, I don't think we can create workarounds for every possible invalid modification of the API.

**Comment by ThomasPe** [comment]
The fix for this issue has been shipped to nuget with Version 1.6.1. Can you please test if it works for you?
